blog.qmd

```diff
@@ -0,0 +1,329 @@
+    the [Florida Department of Education's course catalog
+    website](https://flscns.fldoe.org/Default.aspx). This is the main body of
+    text that will be analyzed using text mining techniques. The text appears to
+    be truncated in the provided data, as indicated by the ellipsis at the end
+    of each string.
+
+```{r}
+courses <-
+  readtext(here::here("data/data-intermediate/course-descriptions.csv"), 
+           text_field = "course_description",
+           docid_field = 'course_id') %>% 
+  janitor::clean_names() %>% 
+  filter(course_status == "ACTIVE",
+         type_of_credit == "COLLEGE CREDIT") %>% 
+  select(doc_id, discipline, course_title, course_credits, text) %>%
+  mutate(course_title = str_to_title(course_title),
+         discipline = str_to_title(discipline),
+         text = str_to_sentence(text))
+```
+
+```{r}
+#| echo: false
+courses %>% 
+  slice(1:5) %>% 
+  gt() %>% 
+    opt_interactive(
+    use_resizers = TRUE,
+    use_filters = TRUE,
+    use_compact_mode = TRUE,
+    use_highlight = TRUE,
+    use_search = TRUE,
+    use_text_wrapping = TRUE
+  ) %>%
+    cols_width(
+    doc_id ~ px(150),
+    discipline ~ px(180),
+    course_title ~ px(150),
+    course_credits ~ px(80),
+    text ~ px(350)
+  ) %>% 
+  urbn_gt_theme()
+```
+
+The first step in our analysis is to create a corpus, a collection of text
+documents, from our course descriptions. We can achieve this using the
+`corpus()` function in `quanteda.` Next, we extract the tokens in the
+corpus---usually words, but they can also be n-grams or multi-word expressions.
+The tokens function allows us to define what we mean by tokens and apply some
+rules to ignore elements such as punctuation and digits.
+
+```{r}
+corpus <- corpus(courses)
+```
+
+The output of this code is a corpus object that consists of the course
+descriptions from our data frame. Each document in the corpus corresponds to a
+course description, and the corpus contains metadata about each document. For
+instance, the first few documents in the corpus look something like this:
+
+```{r}
+#| echo: false
+corpus[1:3]
+```
+
+Each line represents a document in the corpus, with the document ID (e.g.,
+"BC-THE-2300") followed by a snippet of the course description. This corpus will
+serve as the basis for our subsequent text analysis.
+
+After tokenizing the corpus, we obtain a list of tokens for each document. These tokens are essentially the individual words from each course description, stripped of punctuation and numbers.
+```{r}
+tk <- tokens(corpus, what = "word", remove_punct = TRUE, remove_numbers = TRUE)
+```
+
+This function returns a list of tokens for each document in the corpus which looks like this:
+```{r}
+#| echo: false
+tk[1:3]
+```
+
+Each line represents a document in the corpus, with the document ID (e.g., "BC-THE-2300") followed by a list of tokens extracted from the course description. These tokens will be used for our subsequent text analysis tasks.
```

Comment:
Think this might be a redundant line - I see almost the same thing above

Reply:
Re-upping this

Reply:
Removed